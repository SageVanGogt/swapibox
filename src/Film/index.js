import React from 'react';
import PropTypes from 'prop-types';
import './index.css';

const Film = (props) => {
  const { openingCrawl, title, releaseYear } = props.currentRandomFilm;
  return (
    <div className="film-container">
      <img src="/assets/starwarstv.png" alt="tv" className="tv-image" />
      <div className="scrolling-text">
        <p>{openingCrawl}</p>
      </div>
      <h2 className="title">{title}</h2>
      <h4 className="year">{releaseYear}</h4>
    </div>
  );
}

Film.propTypes = {
  currentRandomFilm: PropTypes.object.isRequired
};

export default Film;