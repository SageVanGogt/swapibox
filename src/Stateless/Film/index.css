.scrolling-text {
  grid-area: a;
  height: 125px;
  width: 130px;
  top: 100px;
  left: 130px;
  overflow: hidden;
  background: transparent;
  color: white;
  border: 1px solid black;
  position: absolute;
  -webkit-transform: rotate(-6deg) skew(-6deg, 0);
  -moz-transform: rotate(-6deg) skew(-6deg, 0);
  -ms-transform: rotate(-6deg) skew(-6deg, 0);
  -o-transform: rotate(-6deg) skew(-6deg, 0);
  transform: rotate(-6deg) skew(-6deg, 0); }

.scrolling-text p {
  position: absolute;
  width: 100%;
  height: 100%;
  margin: 0;
  line-height: 20px;
  text-align: center;
  /* Starting position */
  -moz-transform: translateY(100%);
  -webkit-transform: translateY(100%);
  transform: translateY(100%);
  /* Apply animation to this element */
  -moz-animation: scrolling-text 7s linear infinite;
  -webkit-animation: scrolling-text 7s linear infinite;
  animation: scrolling-text 7s linear infinite; }

/* Move it (define the animation) */
@-moz-keyframes scrolling-text {
  0% {
    -moz-transform: translateY(100%); }
  100% {
    -moz-transform: translateY(-100%); } }

@-webkit-keyframes scrolling-text {
  0% {
    -webkit-transform: translateY(100%); }
  100% {
    -webkit-transform: translateY(-100%); } }

@keyframes scrolling-text {
  0% {
    -moz-transform: translateY(100%);
    /* Browser bug fix */
    -webkit-transform: translateY(100%);
    /* Browser bug fix */
    transform: translateY(100%); }
  100% {
    -moz-transform: translateY(-100%);
    /* Browser bug fix */
    -webkit-transform: translateY(-100%);
    /* Browser bug fix */
    transform: translateY(-100%); } }

.tv-image {
  height: 300px;
  position: absolute;
  left: -30px;
  margin: 0;
  padding: 0; }

.title, .year {
  background: none;
  font-family: 'Righteous', 'cursive';
  position: relative;
  color: #ffd54e; }

.year {
  line-height: 5px;
  font-size: 1.5rem; }

.title {
  font-size: 2rem; }
