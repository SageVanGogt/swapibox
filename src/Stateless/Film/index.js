import React from 'react';
import PropTypes from 'prop-types';
import './index.css';

const Film = (props) => {
  const { openingCrawl, title, releaseYear } = props.currentRandomFilm;
  return (
    <div className="film-container">
      <img src="/assets/starwarstv.png" alt="tv" className="tv-image" />
      <div className="scrolling-text">
        <p>{openingCrawl}</p>
      </div>
      <div className="title-year-container">
        <h2 className="title">{title}</h2>
        <h4 className="year">{releaseYear}</h4>
      </div>
    </div>
  );
};

Film.propTypes = {
  currentRandomFilm: PropTypes.object.isRequired
};

export default Film;