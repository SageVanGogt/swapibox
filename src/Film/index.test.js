import React from 'react';
import Film from './index';
import { shallow } from 'enzyme';

describe('Film container component', () => {
  it('should match the snapshot', () => {
    const props = {
      openingCrawl: 'aca', 
      title: 'aca', 
      releaseYear: 'aca'
    }
    const wrapper = shallow(<Film 
      currentRandomFilm={props}
    />);

    console.log(wrapper)

    expect(wrapper).toMatchSnapshot();
  })
})