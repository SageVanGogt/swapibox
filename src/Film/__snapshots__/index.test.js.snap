// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Film container component should match the snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Film
    currentRandomFilm={
        Object {
            "openingCrawl": "aca",
            "releaseYear": "aca",
            "title": "aca",
          }
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <p>
          aca
</p>,
        <h2>
          aca
</h2>,
        <h4>
          aca
</h4>,
      ],
      "className": "film-container",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "aca",
        },
        "ref": null,
        "rendered": "aca",
        "type": "p",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "aca",
        },
        "ref": null,
        "rendered": "aca",
        "type": "h2",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "aca",
        },
        "ref": null,
        "rendered": "aca",
        "type": "h4",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <p>
            aca
</p>,
          <h2>
            aca
</h2>,
          <h4>
            aca
</h4>,
        ],
        "className": "film-container",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "aca",
          },
          "ref": null,
          "rendered": "aca",
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "aca",
          },
          "ref": null,
          "rendered": "aca",
          "type": "h2",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "aca",
          },
          "ref": null,
          "rendered": "aca",
          "type": "h4",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
